### DISCUSSION TITLE
Write all styles inline, is it bad?

### DISCUSSION THREAD

**Discussion by TelezhenkoPV** [bn-digital/home#14] [Q&A] [answered]
We can create a folder for constant and call it 'styles' 
```
export const styles = {
  colorTextPrimary: '#98ab32',
  colorTextSecondary: '#2a6bff'
}

export const sectionHero = {
  fontSize: '1rem',
  color: styles.colorTextPrimary,
  paddingTop: 20
}
```

Then import it in component

`import { styles, sectionHero } from 'src/styles'`

and use 

```
const Demo: FC = () => { 
 const isMobile = useMediaQuery({ maxWidth: 500 })

  return (
      <>
        <Typography.Text style={sectionHero}>Some wonderful text</Typography.Text>
        <Typography.Text style={{ color: isMobile ? styles.colorTextPrimary : styles.colorTextSecondary }}>Some other text</Typography.Text>
      </>
  )
}
```

**Comment by vladyslavvolkov** [comment, marked as answer]
Just imagine refactoring complexity replacing `primary` and `secondary` color (which usually comes within *.less context, as well as changing mobile breakpoint definition.

KISS: 

**Component:**
```tsx
  <Typography.Text className={'ant-typography-my-text'}>Some other text</Typography.Text>
```

**Style:**
```less
.ant-typography {

  &-my-text {
    color: @primary-color;

    @media (max-width: @screen-md) {
      color: @primary-color;
    }
  
}
```

Key points:

- Ant has breakpoints as variables (no hardcode)
- CSS is more performant, will be minimized, no surprises in terms of listener on window.resize event
- No magic for further override and reusability
- variable `sectionHero` will be created **every** component's render
- no linter applied on inline styles
- single responsibility rule break (component  state/behavior & presentation layers messed up)

**Comment by vladyslavvolkov** [comment]
Recommend to check out this [Stackoverflow](https://stackoverflow.com/questions/26882177/react-js-inline-style-best-practices) topic

**Comment by TelezhenkoPV** [comment]
Thanks for your answer, Vlad!
Nice to understand this theme more.

As a side note, [this video](https://www.youtube.com/watch?v=ERB1TJBn32c) shows a more grounded topic of inline styles in JSX.

Main idea is: inline we should add styles, that depends on component state.
